### DISCUSSION TITLE
Roadmap for future ideas

### DISCUSSION THREAD

**Discussion by njames** [tighten/jigsaw#700] [Ideas]
Hey all, 
I am just a fan of Jigsaw as I like to be able to generate sites very easily from markdown files with an easy templating language but less complex than a full CMS - for that I would go to Statmic.

So what features would make Jigsaw super easy to get started and easy to deploy?

I know you have great doco already and a library of templates / sites that use Jigsaw but it would be nice to:
1. link to packagist for the path to starting a site with a library from [Built With Jigsaw](https://builtwithjigsaw.com/)
2. add [prompts](https://github.com/laravel/prompts) to an `jigsaw init`
3. update some content on BWJ and Update the Tighten starter templates
4. Filter BWJ to those that are starter templates

I have been looking at 11ty.dev and that started the thoughts and it would be great for PHP to have a SSR that was at least as good.

Cheers, 
Nigel

**Comment by husseinalhammad** [comment]
I would love a more transparent roadmap for Jigsaw too!

**Comment by wturrell** [comment]
I started using Jigsaw recently and last night I listened to [this podcast](https://twentypercent.fm/episodes/keith-damiani-jigsaw/transcript), if you haven't already. It does talk about the intention to have a Jigsaw 2.0 (with an upgrade path, obviously) but it is nearly three years old now and my presumption is that Tighten are just really busy with other stuff - and it was clear it was going to be a substantial rewrite.

As an enthusiastic user of Statamic too, what I like about Jigsaw is that for a (non-starter template) site, your composer.json is essentially "require tightenco/jigsaw" and, err, that's it. And you still get a working Tailwind out of the box. So its *very* quick to get going and it also runs well on ancient hardware. (Not that Statamic doesn't - it's just - you know, lighter).  And while you can edit markdown content by hand in Statamic, it's not great (imho) for creating new items outside of the UI, because of all the extra front matter, UUIDs etc.

I suppose the obvious wishlist item for Statamic users would be Antlers support as well as Blade. Which has two benefits:

(a) less of the pain of mentally switching between two templating languages
(b) if you build something that outgrows Jigsaw, you can reuse your Antlers templates in Statamic, which makes it easier to decide to go with Jigsaw for those sites where you're not *quite* sure whether you need a full CMS.

Also, personally I'd like to be able to be able to optionally build one file at a time (use case - site where I'm pulling in information from live RSS feeds etc. so there needs to be a regular cron job - and it just feels wasteful rebuilding the entire thing). Not necessarily straightforward though, with dependencies.

**Reply by husseinalhammad** [reply]
I think Jigsaw should stick to Blade only. If a developer / team works on a mix of Laravel/Jigsaw/Statamic projects, I think they should embrace the fact they work with tools from the Laravel ecosystem and stick to Blade even on Statamic, which is possible:

- [`@antlers` directive to write raw Antlers](https://statamic.dev/blade#writing-pure-antlers-in-blade)
- [using Antlers tags with Blade](https://statamic.dev/blade#using-tags-with-blade)
- [using Antlers modifiers with Blade](https://statamic.dev/blade#using-modifiers-with-blade)

Having said that, I agree with you that some convenient growth path for Jigsaw sites could be nice. I think many sites can live with just the below (& that's already possible):
- Jigsaw
- Headless CMS (e.g. Prismic)
- Form handler (e.g. Tighten's [FieldGoal](https://fieldgoal.io/))

But perhaps there is no clear growth path for a site that eventually needs to have a backend.

**Reply by wturrell** [reply]
Yep, good points. I'm gradually coming around to the idea of trying to do more in Blade.

**Comment by njames** [comment]
Thanks for chiming in @wturrell. Actually I started again on another site with jigsaw after looking at laravel/folio and deciding it didn't have the stuff in the box to make it work the way I wanted.

So back to jigsaw and over all I like it. 

In my conversation with Keith D (tech lead at Tighten) they are committed to jigsaw and it is actively used by them. 

I have started looking at adding prompts which got me down this rabbit hole but we got the illuminate packages updated which was a good start. 

Happy days. 

Nigel
